library/include/rocblas_bfloat16.h

```diff
@@ -74,45 +74,58 @@ struct rocblas_bfloat16
     }
 
 private:
-    static constexpr __host__ __device__ uint16_t float_to_bfloat16(float f)
+    static constexpr __host__ __device__ uint16_t float_to_bfloat16(float f, bool round = true)
     {
         union
         {
             float    fp32;
             uint32_t int32;
         } u = {f};
-        if(~u.int32 & 0x7f800000)
+        if(round)
```

Comment:
This logic is incorrect. We do not want it to apply if `~u.int32 & 0x7f800000` returns 0, which means the value is either Inf or NaN.

Why are you modifying `rocblas_bfloat16.h`? It seems like this is better handled in the testing code.

Reply:
The check on ~u.int32 & 0x7f800000 is still there, we just have the option of either rounding or truncating.

I think it makes more sense to have an option to truncate built into the type rather than in some parts of the test code. The operation is specific to this type, and rocblas was recently refactored to remove all independent specialized type conversion code in favor of simpler types and casting.

Reply:
But this rounding is specific to certain kernels, and the only reason we're doing all of this is so that tests for certain sizes pass. As such, the truncation should go into the test code. We already have to modify the test code to account for the different sizes, and take special steps for BF16. We should write a truncation function in `clients/include/rocblas_math.hpp` or even in the same file as the test code, and call it.

We do not have functions for truncating instead of rounding for other types, so we should not introduce them for BF16, since then we have to introduce it for all types if we are to be consistent. Since this is a corner test case where a different kernel behavior is expected, the different rounding/truncation behavior belongs in the test code.

Reply:
Moved the code to float_to_bfloat16_truncate function